lib/ReactViews/Workbench/Controls/Legend.tsx

```diff
@@ -0,0 +1,302 @@
+  "image/x-bmp"
+].reduce<{ [key: string]: boolean }>((acc, mimeType) => {
+  acc[mimeType] = true;
+  return acc;
+}, {});
+const IMAGE_URL_REGEX = /[.\/](png|jpg|jpeg|gif|svg)/i;
+
+function checkMimeType(legend: Model<LegendTraits>) {
+  if (legend.urlMimeType) {
+    return !!DISPLAYABLE_MIME_TYPES[legend.urlMimeType];
+  }
+
+  return !!legend.url?.match(IMAGE_URL_REGEX);
+}
+
+@observer
+export default class Legend extends React.Component<{
+  item: CatalogMemberMixin.CatalogMemberMixin;
+  forPrint?: boolean;
+}> {
+  static defaultProps = {
+    forPrint: false
+  };
+
+  // If we legends need scaling, this is the only way to do it :/
+  // See https://stackoverflow.com/questions/7699621/display-image-at-50-of-its-native-size
+  // or https://stackoverflow.com/questions/35711807/display-high-dpi-image-at-50-scaling-using-just-css
+  resizeLegendImage(
+    evt: SyntheticEvent<HTMLObjectElement>,
+    legend: Model<LegendTraits>
+  ) {
+    if (!isDefined(legend.imageScaling) || legend.imageScaling === 1) return;
+    const image = evt.target as HTMLObjectElement;
+
+    image.style.width = `${legend.imageScaling * image.offsetWidth}px`;
+    // Must set maxWidth *after* setting width, as it may change offsetWidth
+    image.style.maxWidth = "100%";
+  }
+
+  renderLegend(legend: Model<LegendTraits>, i: number) {
+    if (defined(legend.url)) {
+      return this.renderImageLegend(legend, i);
+    } else if (defined(legend.items)) {
+      return this.renderGeneratedLegend(legend, i);
+    }
+    return null;
+  }
+
+  renderImageLegend(legend: Model<LegendTraits>, i: number) {
+    const isImage = checkMimeType(legend);
+    // const insertDirectly = !!legend.safeSvgContent; // we only insert content we generated ourselves, not arbitrary SVG from init files.
+
+    // const svg = legend.safeSvgContent;
+    // // Safari xlink NS issue fix
+    // const processedSvg = svg ? svg.replace(/NS\d+:href/gi, "xlink:href") : null;
+    // const safeSvgContent = { __html: processedSvg };
+
+    /* We proxy the legend so it's cached, and so that the Print/Export feature works with non-CORS servers.
+     * We make it absolute because the print view is opened on a different domain (about:blank) so relative
+     * URLs will not work.
+     */
+    const proxiedUrl = isDefined(legend.url)
+      ? makeAbsolute(proxyCatalogItemUrl(this.props.item, legend.url))
+      : undefined;
+
+    // padding-top: 8px;
+    // padding-bottom: 8px;
+
+    // if (isImage && insertDirectly) {
+    //   return (<li
+    //     key={i}
+    //     className={Styles.legendSvg}
+    //     dangerouslySetInnerHTML={safeSvgContent}
+    //   />)
+    // }
+
+    if (!isDefined(proxiedUrl)) return null;
+
+    if (isImage) {
+      return (
+        <li key={proxiedUrl}>
+          <a
+            href={proxiedUrl}
+            className={Styles.imageAnchor}
+            target="_blank"
+            rel="noreferrer noopener"
+          >
+            <object
```

Comment:
why are we using `object` here instead of `img` when the object is an image?

Reply:
This is so we don't get the broken image icon